libnetwork/endpoint_store_test.go

```diff
@@ -40,8 +40,8 @@ func TestEndpointStore(t *testing.T) {
 	found := c.findEndpoints(filterEndpointByNetworkId("testNetwork"))
 	slices.SortFunc(found, epSorter)
 	assert.Equal(t, len(found), 2)
-	assert.Equal(t, found[0], ep1)
-	assert.Equal(t, found[1], ep2)
+	assert.Check(t, found[0] == ep1, "got: %s; expected: %s", found[0].id, ep1.id)
+	assert.Check(t, found[1] == ep2, "got: %s; expected: %s", found[1].id, ep2.id)
```

Comment:
I think this would work? ...
```suggestion
	assert.Check(t, is.Equal(found[0], ep1))
	assert.Check(t, is.Equal(found[1], ep2))
```

Reply:
I think this was done to use the custom message to prevent the whole `ep` being dumped as "not equal".

Possibly a custom message would work in that case (but not sure if `is.Equal` would still dump the whole `ep`);

```go
	assert.Check(t, is.Equal(found[0], ep1), "got: %s; expected: %s", found[0].id, ep1.id)
```